gitpod-io/gitpod
# Add persistent volume support for workspaces

## Description

Add support for persistent volume claim use for workspaces instead of using local storage.
Usage is gated behind "persistent volume claim" feature flag.

## Related Issue(s)

Fixes #9142
Fixed #9442 

## How to test

Spin up new workspace preview environment. 
go into admin panel and enable feature flag
start new workspace. it will start using dedicated persistent volume.
observe new pvc created (`kubectl get pvc -A`)
delete workspace. pvc will be deleted.

image build is using old codepath and not affected by this change.
prebuilds are using old codepath and are not affected by this change.

limitations:
current implementation doesn't support restoring from previous backup. That will be added separately. 

## Release Notes

```release-note
[experimental] Add persistent volume support for workspaces
```

## Documentation
